package/MDAnalysis/coordinates/RDKit.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""RDKit molecule --- :mod:`MDAnalysis.coordinates.RDKit`
+================================================================
+
+Read coordinates data from an `RDKit <https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html#rdkit.Chem.rdchem.Mol>`_ :class:`rdkit.Chem.rdchem.Mol` with :class:`RDKitReader` 
+into a MDAnalysis Universe. Convert it back to a :class:`rdkit.Chem.rdchem.Mol` with 
+:class:`RDKitConverter`.
+
+
+Example
+-------
+
+>>> from rdkit import Chem
+>>> import MDAnalysis as mda
+>>> mol = Chem.MolFromMol2File("docking_poses.mol2", removeHs=False)
+>>> u = mda.Universe(mol)
+>>> u
+<Universe with 42 atoms>
+>>> u.trajectory
+<RDKitReader with 10 frames of 42 atoms>
+
+
+Classes
+-------
+
+.. autoclass:: RDKitReader
+   :members:
+
+.. autoclass:: RDKitConverter
+   :members:
+
+
+"""
+from __future__ import absolute_import
+
+import numpy as np
+
+from . import memory
+
+
+class RDKitReader(memory.MemoryReader):
+    """Coordinate reader for RDKit.
+    
+    .. versionadded:: 1.0.0
+    """
+    format = 'RDKIT'
+
+    # Structure.coordinates always in Angstrom
+    units = {'time': None, 'length': 'Angstrom'}
+
+    @staticmethod
+    def _format_hint(thing):
+        """Can this reader read *thing*?
+
+        .. versionadded:: 1.0.0
+        """
+        try:
+            from rdkit import Chem
+        except ImportError:
+            # if we can't import rdkit, it's probably not rdkit
+            return False
+        else:
+            return isinstance(thing, Chem.Mol)
+
+    def __init__(self, filename, **kwargs):
+        """Read coordinates from an RDKit molecule.
+        Each conformer in the original RDKit molecule will be read as a frame 
+        in the resulting universe.
+
+        Parameters
+        ----------
+
+        filename : rdkit.Chem.rdchem.Mol
+            RDKit molecule
+        """
+        n_atoms = filename.GetNumAtoms()
+        coordinates = np.array([
+            conf.GetPositions() for conf in filename.GetConformers()], 
+            dtype=np.float32)
+        if coordinates.size == 0:
```

Comment:
Does this happen when the Mol has no coordinates? Worth raising a warning at least.

Reply:
Yes, and since the memory reader expects a numpy array I thought it would be best to have it at least with the right shape

Reply:
Do we have a feeling for what the standard behaviour across the MDA readers is? I believe the netcdf reader will just throw an error if it can't find a reasonable dimension (my thought here is that we would want to have a similar behaviour to how other things work).